plugins/sigma.layout.noverlap/sigma.layout.noverlap.js

```diff
@@ -0,0 +1,408 @@
+          ymin = Infinity,
+          ymax = -Infinity,
+          xwidth,
+          yheight,
+          xcenter,
+          ycenter,
+          grid,
+          row,
+          col,
+          minXBox,
+          maxXBox,
+          minYBox,
+          maxYBox,
+          adjacentNodes,
+          subRow,
+          subCol,
+          nxmin,
+          nxmax,
+          nymin,
+          nymax;
+
+      this.iterCount--;
+      this.running = false;
+
+      for (i=0; i < nodesCount; i++) {
+        n = nodes[i];
+        n.dn.dx = 0;
+        n.dn.dy = 0;
+
+        //Find the min and max for both x and y across all nodes
+        xmin = Math.min(xmin, n.dn_x - (n.dn_size*self.config.scaleNodes + self.config.nodeMargin) );
+        xmax = Math.max(xmax, n.dn_x + (n.dn_size*self.config.scaleNodes + self.config.nodeMargin) );
+        ymin = Math.min(ymin, n.dn_y - (n.dn_size*self.config.scaleNodes + self.config.nodeMargin) );
+        ymax = Math.max(ymax, n.dn_y + (n.dn_size*self.config.scaleNodes + self.config.nodeMargin) );
+
+      }
+
+      xwidth = xmax - xmin;
+      yheight = ymax - ymin;
+      xcenter = (xmin + xmax) / 2;
+      ycenter = (ymin + ymax) / 2;
+      xmin = xcenter - self.config.permittedExpansion*xwidth / 2;
+      xmax = xcenter + self.config.permittedExpansion*xwidth / 2;
+      ymin = ycenter - self.config.permittedExpansion*yheight / 2;
+      ymax = ycenter + self.config.permittedExpansion*yheight / 2;
+
+      grid = {}; //An object of objects where grid[row][col] is an array of node ids representing nodes that fall in that grid. Nodes can fall in more than one grid
+
+      for(row = 0; row < self.config.gridSize; row++) {
+        grid[row] = {};
+        for(col = 0; col < self.config.gridSize; col++) {
+          grid[row][col] = [];
+        }
+      }
+
+      //Place nodes in grid
+      for (i=0; i < nodesCount; i++) {
+        n = nodes[i];
+
+        nxmin = n.dn_x - (n.dn_size*self.config.scaleNodes + self.config.nodeMargin);
+        nxmax = n.dn_x + (n.dn_size*self.config.scaleNodes + self.config.nodeMargin);
+        nymin = n.dn_y - (n.dn_size*self.config.scaleNodes + self.config.nodeMargin);
+        nymax = n.dn_y + (n.dn_size*self.config.scaleNodes + self.config.nodeMargin);
+
+        minXBox = Math.floor(self.config.gridSize* (nxmin - xmin) / (xmax - xmin) );
+        maxXBox = Math.floor(self.config.gridSize* (nxmax - xmin) / (xmax - xmin) );
+        minYBox = Math.floor(self.config.gridSize* (nymin - ymin) / (ymax - ymin) );
+        maxYBox = Math.floor(self.config.gridSize* (nymax - ymin) / (ymax - ymin) );
+        for(col = minXBox; col <= maxXBox; col++) {
+          for(row = minYBox; row <= maxYBox; row++) {
+            grid[row][col].push(n.id);
+          }
+        }
+      }
+
+
+      adjacentNodes = {}; //An object that stores the node ids of adjacent nodes (either in same grid box or adjacent grid box) for all nodes
+
+      for(row = 0; row < self.config.gridSize; row++) {
```

Comment:
Can you test it on a large graph (say at least ~5000 nodes) and tell me whether this works?